src/packagedef.jl

```diff
@@ -1178,15 +1178,15 @@ function get_def(method::Method; modified_files=revision_queue)
         def = get_def(method, pkgdata, file)
         def !== nothing && return true
     end
-    @warn "$(method.sig) was not found"
+    @warn "$(method.sig)$(isdefined(method, :external_mt) ? " (overlayed)" : "") was not found"
     # So that we don't call it again, store missingness info in CodeTracking
-    CodeTracking.invoked_setindex!(CodeTracking.method_info, method.sig, missing)
+    CodeTracking.invoked_setindex!(CodeTracking.method_info, CodeTracking.method_info_key(method), missing)
     return false
 end
 
 function get_def(method, pkgdata, filename)
     maybe_extract_sigs!(maybe_parse_from_cache!(pkgdata, filename))
-    return get(CodeTracking.method_info, method.sig, nothing)
+    return get(CodeTracking.method_info, CodeTracking.method_info_key(method), nothing)
```

Comment:
Perhaps we should add `method_info_key` to `using CodeTracking: ...`

It's slightly idiosyncratic, but I like the module-scoping of `CodeTracking.method_info` because it emphasizes who keeps the data. But `method_info_key` seems like we could just make it a known function in the Revise namespace. But if you aren't a fan of this proposal, it's fine the way it is.

Reply:
That makes perfect sense, I like this style.